app/main.dev.js

```diff
@@ -121,21 +139,7 @@ const handleSlippiURIAsync = async (url) => {
 
     // Wait until mainWindow exists so that we can send an IPC to play.
     // We are willing to wait for a few seconds before timing out
-    const wait = ms => new Promise((resolve) => setTimeout(resolve, ms));
-    let retryIdx = 0;
-    while (!didFinishLoad && retryIdx < 200) {
-      // It's okay to await in loop, we want things to be slow in this case
-      await wait(100); // eslint-disable-line
-      retryIdx += 1;
-    }
-
-    if (retryIdx === 100) {
-      log.warn("Timed out waiting for mainWindow to exist.");
-      return;
-    }
-
-    log.info(`Found mainWindow after ${retryIdx} tries.`);
-    mainWindow.webContents.send("play-replay");
+    waitForMainWindow().then(() => { mainWindow.webContents.send("play-replay", url); });  // eslint-disable-line
```

Comment:
This actually changes the functionality somewhat. Previously if waiting for the main window timed out, we would not try to play the replay, with the way it's written now, the send will still occur. The timeout should probably be converted to an error inside the promise function.